plugins/github/api.ml

```diff
@@ -522,6 +568,102 @@ let head_of t repo id =
 
+module CheckRun = struct
+  module Set_status = struct
+    let id = "github-check-run-set-status"
+
+    type nonrec t = t
+
+    module Key = struct
+      type t = {
+          commit : Commit_id.t;
+          context : string; (** TODO Make this app_id? *)
+        }
+
+      let to_json { commit; context } =
+        `Assoc [
+            "commit", `String (Commit_id.digest commit);
+            "context", `String context
+          ]
+
+      let digest t = Yojson.Safe.to_string (to_json t)
+    end
+
+    module Value = CheckRunStatus
+
+    module Outcome = Current.Unit
+
+    let auto_cancel = true
+
+    let pp f ({ Key.commit; context }, status) =
+      Fmt.pf f "Set %a/%s to@ %a"
+        Commit_id.pp commit
+        context
+        Value.pp status
+
+    let publish t job key status =
+      Current.Job.start job ~pool ~level:Current.Level.Above_average >>= fun () ->
+      get_token t >>= function
+      | Error (`Msg m) -> Lwt.fail_with m
+      | Ok token ->
+         let token = Github.Token.of_string token in
+         let (owner, repo) = Commit_id.name key.Key.commit in
+         let sha = key.Key.commit.hash in
+         let app_id = key.Key.context in (* TODO Thread this app_id config in here! *)
+         let check_name = "ocurrent" in (* TODO Should be application name as we want it to appear in GH UI. *)
+
+         let create_check () =
+           let open Github in
+           let body = `Assoc (("name", `String check_name) 
+                              :: ("head_sha", `String key.Key.commit.hash)
+                              :: Value.json_items status) |> Yojson.Safe.to_string in
+           Log.info (fun f -> f "create_check: %s" body);
+           Check.create_check_run ~token ~owner ~repo ~body () in
+
+         let fetch_check_run () =
+           let open Github in
+           let open Monad in
+           (* Assuming a single check_run per app per sha per check_name. *)
+           Check.list_check_runs_for_ref ~token ~owner ~repo ~sha ~app_id ~check_name () >>~ 
+           fun l -> return @@ List.nth_opt l.check_runs 1 in
+
+         let update_check_run (check_run : Github_j.check_run) () =
+           let open Github in
+           let check_run_id = Int64.to_string check_run.check_run_id in
+           let body = `Assoc (Value.json_items status) |> Yojson.Safe.to_string in
+           Log.info (fun f -> f "update_check: %s" body);
+           Check.update_check_run ~owner ~repo ~check_run_id ~body () in
+
+         Lwt.try_bind ( fun () ->
+             let open Github in
+             let open Monad in                        
+             run (
+               fetch_check_run () >>= function
+               | None -> create_check ()
+               | Some check_run -> update_check_run check_run ()))
+
+           (* Ignore the response and return unit. We could return the check_run? *)
+           (fun _ -> Lwt_result.return ())
```

Comment:
```suggestion
           (fun (_ : string) -> Lwt_result.return ())
```
(to avoid problems with partial application if the GitHub API changes to take an extra argument in future)